src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -2015,6 +2015,16 @@ QsCallable GetCallable((QsQualifiedName, ImmutableArray<QsSpecialization>) specI
                 }
                 compilation.UpdateCallables(callables);
                 compilation.UpdateTypes(types);
+
+                var allCallableDeclarations = compilation.GlobalSymbols.DefinedCallables().ToImmutableDictionary(x => x.QualifiedName);
```

Comment:
Instead of creating this dictionary, can you use `compilation.GetCallables()` instead?

Reply:
The info we want is the position and source file of each callable, which we are getting from the declaration headers. We might be able to get this off the callables if those fields are defined on them. I'll look into it.

Reply:
Looks like that works well.